snap/local/hooks/cmd/configure/configure.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package main
+
+import (
+	"os"
+	"strings"
+
+	"github.com/canonical/edgex-snap-hooks/v2/log"
+	"github.com/canonical/edgex-snap-hooks/v2/options"
+	"github.com/canonical/edgex-snap-hooks/v2/snapctl"
+)
+
+func main() {
+	log.SetComponentName("configure")
+	err := options.ProcessAppConfig("device-virtual")
+	if err != nil {
+		log.Errorf("could not process options: %v", err)
+		os.Exit(1)
+	}
+
+	log.Info("Enabling config options")
+	err = snapctl.Set("app-options", "true").Run()
+	if err != nil {
+		panic(err)
+	}
```

Comment:
This should be done before the processing. Otherwise, when the snap is seeded with config options, the first call to the config hook will not have the app-options set to true.

Also, don't `panic`. Log the error and exit with code 1.